lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -42,9 +46,7 @@ class MysqliStatement implements \IteratorAggregate, Statement
         ParameterType::BOOLEAN      => 'i',
         ParameterType::NULL         => 's',
         ParameterType::INTEGER      => 'i',
-
-        // TODO Support LOB bigger then max package size
```

Comment:
@mpdude the fact that this is done is not covered by tests.

Reply:
@mpdude isn't using `send_long_data()` the way to store the data longer than the max package size?

Reply:
No, I tried to explain that in the comments and discussion. You *cannot* overcome the `max_allowed_package` limitation. 

`send_long_data()` will – for each invocation – send out a chunk of data immediately to the MySQL server. So in the first place, it saves you from having to buffer one big query on the client side (memory allocation) before sending that out. We can read streams in small steps and send data to the server, never keeping everything at once in memory.

The MySQL server will still enforce the `max_allowed_packet` limitation and reject queries exceeding it. That happens even if you have sent more data than allowed in small chunks. 

It's not obvious, but the documentation and also the bug report I linked above shortly mention it.